### DISCUSSION TITLE
How to use pattern syntax highlighting

### DISCUSSION THREAD

**Discussion by p4nj** [WerWolv/ImHex#1167] [Get Help] [answered]
I am new ImHex user and did not found in documentation how to appl pattern on windows. I tried open .dll and .zip file.
I tried latest release and latest nightly build. And cannot create image like shown on first doc page (https://docs.werwolv.net/imhex/)

![2023-06-29 11_16_17-ImHex - libsystre-0 dll (Read Only)](https://github.com/WerWolv/ImHex/assets/25669688/60aaeb6f-99c8-400a-8a19-7c4248c72da9)

**Comment by paxcut** [comment, marked as answer]
What you are looking for is how to load pattern files that can be run on the given input file. For some types of files ImHex is able to determine that for you and will ask you if you want to load some pattern file. In other cases it is up to you to select and load a pattern file. A lot of different types of files are loaded through the import menu. Open File>Import and select menu item called "Pattern file". ImHex should then present you a list of pattern files (.hexpat) to choose from. The name of the file corresponds to the type of file you want to run the pattern at. Since you mention a dll you probably are looking to use either the pe.hexpat or the coff.hexpat. Similarly for a zip file you would select the pattern file called zip.hexpat. 

After loading the file its contents should be displayed on the right window called the pattern editor window. This is where you write or modify pattern files and also where you run them. Click the green play icon found on the bottom row and the pattern should be executed

**Comment by p4nj** [comment]
Thanks, that is feature I was looking for. 
Although coff.hexpat is not workging for me, gives me:
```
E: error[E0004]: Type error.
E:   --> <Source Code>:185:33
E: 185 |     u8 rawData[sizeOfRawData] @ pointerToRawData [[sealed]];
E:                                       ^
E:                                       Array expanded past end of the data.
E: 
E: 
```

**Comment by paxcut** [comment]
What input file are you trying to analyze? Did you try pe pattern instead of coff? I think pe is more developed and has less problems.

**Reply by p4nj** [reply]
coff.hexpat

**Comment by paxcut** [comment]
that's the pattern file. I need to know which exe/dll you are using it on so i can test it and possibly fix, but i suggest you use pe.hexpat instead.

**Comment by p4nj** [comment]
Input does not matter. I tried several: small .c program I compiled but the same promlem occurs if you try with C:\Windows\write.exe.

**Comment by paxcut** [comment]
In the user/appsdata/local/imhex folder you will find a directory called tests. inside there is a patters folder and inside it a test_data folder. There you will find example input files for most of the patterns supported and the all should work. In particular the file called coff.obj is to be used with coff.hexpat and it runs without errors. There are subtle differences between pe and coff format. for exe/dll the recomended one is pe. coff may be for static libraries and obj files but i'm not sure. Note that i am not saying that the formats are defined for those types , i am merely pointing how the imhex patterns were created.

**Reply by p4nj** [reply]
I did not find that folder between folder ..\scripts and ..\themes. Maybe becauese I installed binary app?
I also tried static .lib you mention but no success.

**Comment by paxcut** [comment]
did you check the users appsdata folder and the installation folder. it must be in one of the two.

**Reply by p4nj** [reply]
Doublechecked. No folder named tests comes with windows installer.

**Comment by paxcut** [comment]
you may be right about not being distributed. go to pattern github repo and download/clone it. (https://github.com/WerWolv/ImHex-Patterns)

**Comment by paxcut** [comment]
another possibilityis the content store but i did not see them there either
